# BELLABEAT CASE STUDY

**Scenario**

You are a junior data analyst working on the marketing analyst team at Bellabeat, a high-tech manufacturer of health-focused products for women. Bellabeat is a successful small company, but they have the potential to become a larger player in the global smart device market. Urška Sršen, cofounder and Chief Creative Officer of Bellabeat, believes that analyzing smart device fitness data could help unlock new growth opportunities for the company. You have been asked to focus on one of Bellabeat’s products and analyze smart device data to gain insight into how consumers are using their smart devices. The insights you discover will then help guide marketing strategy for the company. You will present your analysis to the Bellabeat executive team along with your high-level recommendations for Bellabeat’s marketing strategy.

**Questions to Answer**

1.	What are some trends in smart device usage? 
2.	How could these trends apply to Bellabeat customers? 
3.	How could these trends help influence Bellabeat marketing strategy?



## About the Dataset

This dataset generated by respondents to a distributed survey via Amazon Mechanical Turk between 03.12.2016-05.12.2016. Thirty eligible Fitbit users consented to the submission of personal tracker data, including minute-level output for physical activity, heart rate, and sleep monitoring. Individual reports can be parsed by export session ID (column A) or timestamp (column B). Variation between output represents use of different types of Fitbit trackers and individual tracking behaviors / preferences.

Dataset: https://www.kaggle.com/datasets/arashnic/fitbit/data

In [1]:
#import libraries

import os

import pandas as pd
import numpy as np

import warnings
warnings.filterwarnings("ignore")

In [8]:
# read csv files # 1: Mar-Apr, 2: Apr-May

daily_activity_1 = pd.read_csv('fitbit_3.12.16-4.11.16/dailyActivity_merged_mar_apr.csv')
heartrate_seconds_1 = pd.read_csv('fitbit_3.12.16-4.11.16/heartrate_seconds_merged_mar_apr.csv')
hourly_calories_1 = pd.read_csv('fitbit_3.12.16-4.11.16/hourlyCalories_merged_mar_apr.csv')
hourly_intensities_1 = pd.read_csv('fitbit_3.12.16-4.11.16/hourlyIntensities_merged_mar_apr.csv')
hourly_steps_1 = pd.read_csv('fitbit_3.12.16-4.11.16/hourlySteps_merged_mar_apr.csv')
minute_calories_1 = pd.read_csv('fitbit_3.12.16-4.11.16/minuteCaloriesNarrow_merged_mar_apr.csv')
minute_intensities_1 = pd.read_csv('fitbit_3.12.16-4.11.16/minuteIntensitiesNarrow_merged_mar_apr.csv')
minute_met_1 = pd.read_csv('fitbit_3.12.16-4.11.16/minuteMETsNarrow_merged_mar_apr.csv')
minute_sleep_1 = pd.read_csv('fitbit_3.12.16-4.11.16/minuteSleep_merged_mar_apr.csv')
minute_steps_1 = pd.read_csv('fitbit_3.12.16-4.11.16/minuteStepsNarrow_merged_mar_apr.csv')
weight_log_info_1 = pd.read_csv('fitbit_3.12.16-4.11.16/weightLogInfo_merged_mar_apr.csv')



daily_activity_2 = pd.read_csv('fitbit_4.12.16-5.12.16/dailyActivity_merged_apr_may.csv')
heartrate_seconds_2 = pd.read_csv('fitbit_4.12.16-5.12.16/heartrate_seconds_merged_apr_may.csv')
hourly_calories_2 = pd.read_csv('fitbit_4.12.16-5.12.16/hourlyCalories_merged_apr_may.csv')
hourly_intensities_2 = pd.read_csv('fitbit_4.12.16-5.12.16/hourlyIntensities_merged_apr_may.csv')
hourly_steps_2 = pd.read_csv('fitbit_4.12.16-5.12.16/hourlySteps_merged_apr_may.csv')
minute_calories_2 = pd.read_csv('fitbit_4.12.16-5.12.16/minuteCaloriesNarrow_merged_apr_may.csv')
minute_intensities_2 = pd.read_csv('fitbit_4.12.16-5.12.16/minuteIntensitiesNarrow_merged_apr_may.csv')
minute_met_2 = pd.read_csv('fitbit_4.12.16-5.12.16/minuteMETsNarrow_merged_apr_may.csv')
minute_sleep_2 = pd.read_csv('fitbit_4.12.16-5.12.16/minuteSleep_merged_apr_may.csv')
minute_steps_2 = pd.read_csv('fitbit_4.12.16-5.12.16/minuteStepsNarrow_merged_apr_may.csv')
weight_log_info_2 = pd.read_csv('fitbit_4.12.16-5.12.16/weightLogInfo_merged_apr_may.csv')


In [15]:
#Check the shape, info, columns, null values and first 5 rows of each dataframe

dfs = [daily_activity_1, heartrate_seconds_1, hourly_calories_1, hourly_intensities_1, hourly_steps_1, minute_calories_1, minute_intensities_1, minute_met_1, minute_sleep_1, minute_steps_1, weight_log_info_1,
daily_activity_2, heartrate_seconds_2, hourly_calories_2, hourly_intensities_2, hourly_steps_2, minute_calories_2, minute_intensities_2, minute_met_2, minute_sleep_2, minute_steps_2, weight_log_info_2]

df_names = ['daily_activity_1', 'heartrate_seconds_1', 'hourly_calories_1', 'hourly_intensities_1', 'hourly_steps_1', 'minute_calories_1', 'minute_intensities_1', 'minute_met_1', 'minute_sleep_1', 'minute_steps_1', 'weight_log_info_1',
'daily_activity_2', 'heartrate_seconds_2', 'hourly_calories_2', 'hourly_intensities_2', 'hourly_steps_2', 'minute_calories_2', 'minute_intensities_2', 'minute_met_2', 'minute_sleep_2', 'minute_steps_2', 'weight_log_info_2']

# Iterate over the list of DataFrame names and DataFrames
for name, df in zip(df_names, dfs):
    print(f"DataFrame '{name}':")
    print(f"Shape: {df.shape}")
    print(f"Info: {df.info()}")
    print(f"Columns: {df.columns}")
    print(f"Null Values: {df.isnull().sum()}")
    print(f"First 5 rows: {df.head()}")
    print()

DataFrame 'daily_activity_1':
Shape: (457, 15)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 457 entries, 0 to 456
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        457 non-null    int64  
 1   ActivityDate              457 non-null    object 
 2   TotalSteps                457 non-null    int64  
 3   TotalDistance             457 non-null    float64
 4   TrackerDistance           457 non-null    float64
 5   LoggedActivitiesDistance  457 non-null    float64
 6   VeryActiveDistance        457 non-null    float64
 7   ModeratelyActiveDistance  457 non-null    float64
 8   LightActiveDistance       457 non-null    float64
 9   SedentaryActiveDistance   457 non-null    float64
 10  VeryActiveMinutes         457 non-null    int64  
 11  FairlyActiveMinutes       457 non-null    int64  
 12  LightlyActiveMinutes      457 non-null    int64  
 13  SedentaryMinutes  

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1445040 entries, 0 to 1445039
Data columns (total 3 columns):
 #   Column          Non-Null Count    Dtype 
---  ------          --------------    ----- 
 0   Id              1445040 non-null  int64 
 1   ActivityMinute  1445040 non-null  object
 2   Intensity       1445040 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 33.1+ MB
Info: None
Columns: Index(['Id', 'ActivityMinute', 'Intensity'], dtype='object')
Null Values: Id                0
ActivityMinute    0
Intensity         0
dtype: int64
First 5 rows:            Id         ActivityMinute  Intensity
0  1503960366  3/12/2016 12:00:00 AM          0
1  1503960366  3/12/2016 12:01:00 AM          0
2  1503960366  3/12/2016 12:02:00 AM          0
3  1503960366  3/12/2016 12:03:00 AM          0
4  1503960366  3/12/2016 12:04:00 AM          0

DataFrame 'minute_met_1':
Shape: (1445040, 3)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1445040 entries, 0 to 1445039
Data column

Null Values: Id       0
Time     0
Value    0
dtype: int64
First 5 rows:            Id                  Time  Value
0  2022484408  4/12/2016 7:21:00 AM     97
1  2022484408  4/12/2016 7:21:05 AM    102
2  2022484408  4/12/2016 7:21:10 AM    105
3  2022484408  4/12/2016 7:21:20 AM    103
4  2022484408  4/12/2016 7:21:25 AM    101

DataFrame 'hourly_calories_2':
Shape: (22099, 3)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22099 entries, 0 to 22098
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Id            22099 non-null  int64 
 1   ActivityHour  22099 non-null  object
 2   Calories      22099 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 518.1+ KB
Info: None
Columns: Index(['Id', 'ActivityHour', 'Calories'], dtype='object')
Null Values: Id              0
ActivityHour    0
Calories        0
dtype: int64
First 5 rows:            Id           ActivityHour  Calories
0  1503960366  4/12/2016 12:00

Information:
___________________

daily_activity_1 : 457 Rows, 15 columns
No null values, Issue with Data Type ('ActivityDate' to date time)

daily_activity_2 : 940 Rows, 15 columns
No null values, Issue with Data Type ('ActivityDate' to date time)

----------

heartrate_seconds_1 : 1154681 rows, 3 columns
No null values, Issue with Data Type ('Time' to date time)

heartrate_seconds_2 : 2483658 rows, 3 columns
No null values, Issue with Data Type ('Time' to date time)

------------

hourly_calories_1 : 24084 rows, 3 columns
No null values, Issue with Data Type ( 'ActivityHour' tp date time)

hourly_calories_2 : 22099 rows, 3 columns
No null values, Issue with Data Type ( 'ActivityHour' tp date time)

------------

hourly_intensities_2 : 24084 rows, 4 columns
no null values, Issue Data Types ('ActivityHour' to date time)

hourly_intensities_2 : 22099 rows, 4 columns
no null values, Issue Data Types ('ActivityHour' to date time)

-------------

hourly_steps_1 : 24084 rows, 3 columns
no null values, Issue Data Types ('ActivityHour' to date time)

hourly_steps_2 : 22099 rows, 3 columns
no null values, Issue Data Types ('ActivityHour' to date time)

---------------

minute_calories_1 : 1445040 rows, 3 columns
no null values, Issue Data Types ('ActivityMinute' to date time)

minute_calories_2 : 1325580 rows, 3 columns
no null values, Issue Data Types ('ActivityMinute' to date time)

---------------

minute_intensities_1 : 1445040 rows, 3 columns
no null values, Issue Data Types ('ActivityMinute' to date time)

minute_intensities_2 : 1325580 rows, 3 columns
no null values, Issue Data Types ('ActivityMinute' to date time)

----------------

minute_met_1 : 1445040 rows, 3 columns
no null values, Issue Data Types ('ActivityMinute' to date time)

minute_met_2 : 1325580 rows, 3 columns
no null values, Issue Data Types ('ActivityMinute' to date time)

----------------

minute_sleep_1 : 198559 rows, 4 columns
no null values, Issue Data Types ('date' to date time)

minute_sleep_2 : 188521 rows, 4 columns
no null values, Issue Data Types ('date' to date time)

----------------

minute_steps_1 : 1445040 rows, 3 columns
no null values, Issue Data Types ('ActivityMinute' to date time)

minute_steps_2 : 1325580 rows, 3 columns
no null values, Issue Data Types ('ActivityMinute' to date time)

-----------------

weight_log_info_1 : 33 rows, 8 columns
31 null values in 'Fat' column, Issue Data Types ('Date' to date time)

weight_log_info_2 : 67 rows, 8 columns
65 null values in 'Fat' column, Issue Data Types ('Date' to date time)

### Data Manipulation and Transformation

#### Unique Ids

In [109]:
#Total Number of unique ids

print(daily_activity_1['Id'].nunique())
print(daily_activity_2['Id'].nunique())

35
33


In [108]:
#check for unique ids that are in daily_activity_1 but not in daily_activity_2

ids_daily_activity_2 = set(daily_activity_2['Id'].unique())
filtered_daily_activity_1 = daily_activity_1[~daily_activity_1['Id'].isin(ids_daily_activity_2)]
filtered_daily_activity_1['Id'].unique()


array([2891001357, 6391747486], dtype=int64)

There are 35 unique ids

### Merge the datasets

In [16]:
# Merge the datasets for March-April and April-May

daily_activity = pd.concat([daily_activity_1, daily_activity_2], axis=0)
heartrate_seconds = pd.concat([heartrate_seconds_1, heartrate_seconds_2], axis=0)
hourly_calories = pd.concat([hourly_calories_1, hourly_calories_2], axis=0)
hourly_intensities = pd.concat([hourly_intensities_1, hourly_intensities_2], axis=0)
hourly_steps = pd.concat([hourly_steps_1, hourly_steps_2], axis=0)
minute_calories = pd.concat([minute_calories_1, minute_calories_2], axis=0)
minute_intensities = pd.concat([minute_intensities_1, minute_intensities_2], axis=0)
minute_met = pd.concat([minute_met_1, minute_met_2], axis=0)
minute_sleep = pd.concat([minute_sleep_1, minute_sleep_2], axis=0)
minute_steps = pd.concat([minute_steps_1, minute_steps_2], axis=0)
weight_log_info = pd.concat([weight_log_info_1, weight_log_info_2], axis=0)

In [22]:
#check if dataframes are properly merged

merged_dfs = [daily_activity, heartrate_seconds, hourly_calories, hourly_intensities, hourly_steps, minute_calories, minute_intensities, minute_met, minute_sleep, minute_steps, weight_log_info]

merged_df_names = ['daily_activity', 'heartrate_seconds', 'hourly_calories', 'hourly_intensities', 'hourly_steps', 'minute_calories', 'minute_intensities', 'minute_met', 'minute_sleep', 'minute_steps', 'weight_log_info']

# Iterate over the list of DataFrame names and DataFrames
for name, df in zip(merged_df_names, merged_dfs):
    print(f"DataFrame '{name}':")
    print(f"Shape: {df.shape}")
    print(f"Columns: {df.columns}")
    print(f"Null Values: {df.isnull().sum()}")
    print()

DataFrame 'daily_activity':
Shape: (1397, 15)
Columns: Index(['Id', 'ActivityDate', 'TotalSteps', 'TotalDistance', 'TrackerDistance',
       'LoggedActivitiesDistance', 'VeryActiveDistance',
       'ModeratelyActiveDistance', 'LightActiveDistance',
       'SedentaryActiveDistance', 'VeryActiveMinutes', 'FairlyActiveMinutes',
       'LightlyActiveMinutes', 'SedentaryMinutes', 'Calories'],
      dtype='object')
Null Values: Id                          0
ActivityDate                0
TotalSteps                  0
TotalDistance               0
TrackerDistance             0
LoggedActivitiesDistance    0
VeryActiveDistance          0
ModeratelyActiveDistance    0
LightActiveDistance         0
SedentaryActiveDistance     0
VeryActiveMinutes           0
FairlyActiveMinutes         0
LightlyActiveMinutes        0
SedentaryMinutes            0
Calories                    0
dtype: int64

DataFrame 'heartrate_seconds':
Shape: (3638339, 3)
Columns: Index(['Id', 'Time', 'Value'], dtype='object')
Nul

Information
----------------

**daily_activity:** 1397 rows, 15 columns, no null values, data type issues (ActivityDate to datetime)

---------------

**heartrate_seconds:** 3638339 rows, 3 columns, no null values, data type issues (Time to datetime)

----------------

**hourly_calories:** 46183 rows, 3 columns, no null values, data type issues (ActivityHour to datetime)

---------------

**hourly_intensities:** 46183 rows, 4 columns, no null values, data type issues (ActivityHour to datetime)

-----------------

**hourly_steps:** 46183 rows, 3 columns, no null values, data type issues (ActivityHour to datetime)

------------------

**minute_calories:** 2770620 rows, 3 columns, no null values, data type issues (ActivityMinute to datetime)

-------------------

**minute_intensities:** 2770620 rows, 3 columns, no null values, data type issues (ActivityMinute to datetime)

--------------------

**minute_met:** 2770620 rows, 3 columns, no null values, data type issues (ActivityMinute to datetime)

---------------------


**minute_sleep:** 387080 rows, 4 columns, no null values, data type issues (date to datetime)

---------------------

**minute_steps:**  2770620 rows, 3 columns, no null values, data type issues (ActivityMinute to datetime)

--------------------

**weight_log_info:** 100 rows, 8 columns, 96 null values in "Fat", data type issues ( Date to datetime)

# Data Cleaning

### Dealing with null values

In [30]:
#dealing with null values

weight_log_info[weight_log_info['Fat'].isnull()]

Id                   Date    WeightKg  WeightPounds  Fat  \
1   1927972279   4/10/2016 6:33:26 PM  129.600006    285.719105  NaN   
3   2873212765   4/6/2016 11:59:59 PM   56.700001    125.002104  NaN   
4   2873212765   4/7/2016 11:59:59 PM   57.200001    126.104416  NaN   
5   2891001357   4/5/2016 11:59:59 PM   88.400002    194.888643  NaN   
6   4445114986  3/30/2016 11:59:59 PM   92.400002    203.707134  NaN   
..         ...                    ...         ...           ...  ...   
62  8877689391    5/6/2016 6:43:35 AM   85.000000    187.392923  NaN   
63  8877689391    5/8/2016 7:35:53 AM   85.400002    188.274775  NaN   
64  8877689391    5/9/2016 6:39:44 AM   85.500000    188.495234  NaN   
65  8877689391   5/11/2016 6:51:47 AM   85.400002    188.274775  NaN   
66  8877689391   5/12/2016 6:42:53 AM   84.000000    185.188300  NaN   

          BMI  IsManualReport          LogId  
1   46.169998           False  1460313206000  
3   21.450001            True  1459987199000  
4   21.650000            True  1460073599000  
5   25.030001            True  1459900799000  
6   35.009998            True  1459382399000  
..        ...             ...            ...  
62  25.440001           False  1462517015000  
63  25.559999           False  1462692953000  
64  25.610001           False  1462775984000  
65  25.559999           False  1462949507000  
66  25.139999           False  1463035373000  

[96 rows x 8 columns]

Since this data do not provide info about how consumers use smart devices, we are not going to use it for our analysis.

### Change data types

In [35]:
#change data types

daily_activity['ActivityDate'] = pd.to_datetime(daily_activity['ActivityDate'])


In [36]:
heartrate_seconds['Time'] = pd.to_datetime(heartrate_seconds['Time'])


In [37]:
hourly_calories['ActivityHour'] = pd.to_datetime(hourly_calories['ActivityHour'])


In [38]:
hourly_intensities['ActivityHour'] = pd.to_datetime(hourly_intensities['ActivityHour'])


In [39]:
hourly_steps['ActivityHour'] = pd.to_datetime(hourly_steps['ActivityHour'])


In [41]:
minute_calories['ActitvityMinute'] = pd.to_datetime(minute_calories['ActivityMinute'])


In [42]:
minute_intensities['ActitvityMinute'] = pd.to_datetime(minute_intensities['ActivityMinute'])


In [43]:
minute_met['ActitvityMinute'] = pd.to_datetime(minute_met['ActivityMinute'])


In [44]:
minute_sleep['date'] = pd.to_datetime(minute_sleep['date'])


In [45]:
minute_steps['ActitvityMinute'] = pd.to_datetime(minute_steps['ActivityMinute'])

## Reset Index

In [72]:
#Reset Index

merged_dfs = [daily_activity, heartrate_seconds, hourly_calories, hourly_intensities, hourly_steps, minute_calories, minute_intensities, minute_met, minute_sleep, minute_steps, weight_log_info]

# Iterate over the list of DataFrames and reset index in place
for df in merged_dfs:
    df.reset_index(drop=True, inplace=True)

# each DataFrame in merged_dfs has its index reset


In [110]:
daily_activity['Id'].nunique()

35

There are 35 Unique Ids in merged dataset as well.

### Merge Hour and Minute Dataframes

In [154]:
#merge hourly dataframes

hourly_cal_int_step = pd.concat([hourly_calories, hourly_steps['StepTotal'], hourly_intensities[['TotalIntensity', 'AverageIntensity']]], axis=1)

In [156]:
hourly_cal_int_step.head(10)

Id        ActivityHour  Calories  StepTotal  TotalIntensity  \
0  1503960366 2016-03-12 00:00:00        48          0               0   
1  1503960366 2016-03-12 01:00:00        48          0               0   
2  1503960366 2016-03-12 02:00:00        48          0               0   
3  1503960366 2016-03-12 03:00:00        48          0               0   
4  1503960366 2016-03-12 04:00:00        48          0               0   
5  1503960366 2016-03-12 05:00:00        48          0               0   
6  1503960366 2016-03-12 06:00:00        48          0               0   
7  1503960366 2016-03-12 07:00:00        48          0               0   
8  1503960366 2016-03-12 08:00:00        48          0               0   
9  1503960366 2016-03-12 09:00:00        49          8               1   

   AverageIntensity  
0          0.000000  
1          0.000000  
2          0.000000  
3          0.000000  
4          0.000000  
5          0.000000  
6          0.000000  
7          0.000000  
8          0.000000  
9          0.016667

In [157]:
hourly_cal_int_step.shape

(46183, 6)

Succesfully merged.

In [210]:
# Merge minute dataframes except sleep(shape is different)

min_cal_int_met_steps = pd.concat([minute_calories, minute_intensities['Intensity'], minute_met['METs'],minute_steps['Steps']], axis=1)

In [211]:
min_cal_int_met_steps.head(10)

Id         ActivityMinute  Calories     ActitvityMinute  Intensity  \
0  1503960366  3/12/2016 12:00:00 AM    0.7973 2016-03-12 00:00:00          0   
1  1503960366  3/12/2016 12:01:00 AM    0.7973 2016-03-12 00:01:00          0   
2  1503960366  3/12/2016 12:02:00 AM    0.7973 2016-03-12 00:02:00          0   
3  1503960366  3/12/2016 12:03:00 AM    0.7973 2016-03-12 00:03:00          0   
4  1503960366  3/12/2016 12:04:00 AM    0.7973 2016-03-12 00:04:00          0   
5  1503960366  3/12/2016 12:05:00 AM    0.7973 2016-03-12 00:05:00          0   
6  1503960366  3/12/2016 12:06:00 AM    0.7973 2016-03-12 00:06:00          0   
7  1503960366  3/12/2016 12:07:00 AM    0.7973 2016-03-12 00:07:00          0   
8  1503960366  3/12/2016 12:08:00 AM    0.7973 2016-03-12 00:08:00          0   
9  1503960366  3/12/2016 12:09:00 AM    0.7973 2016-03-12 00:09:00          0   

   METs  Steps  
0    10      0  
1    10      0  
2    10      0  
3    10      0  
4    10      0  
5    10      0  
6    10      0  
7    10      0  
8    10      0  
9    10      0

In [165]:
min_cal_int_met_steps.shape

(2770620, 7)

Successfully Merged.

# Final Dataframes for analysis:

-- daily_activity

-- heartrate_seconds

-- hourly_cal_int_step

-- min_cal_int_met_steps

-- minute_sleep

-- wight_log_info

# Daily Activity

In [47]:
daily_activity.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1397 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   Id                        1397 non-null   int64         
 1   ActivityDate              1397 non-null   datetime64[ns]
 2   TotalSteps                1397 non-null   int64         
 3   TotalDistance             1397 non-null   float64       
 4   TrackerDistance           1397 non-null   float64       
 5   LoggedActivitiesDistance  1397 non-null   float64       
 6   VeryActiveDistance        1397 non-null   float64       
 7   ModeratelyActiveDistance  1397 non-null   float64       
 8   LightActiveDistance       1397 non-null   float64       
 9   SedentaryActiveDistance   1397 non-null   float64       
 10  VeryActiveMinutes         1397 non-null   int64         
 11  FairlyActiveMinutes       1397 non-null   int64         
 12  LightlyActiveMinutes 

#### Create Columns for Day_of_the_week and Number_of_week

In [59]:
#Create columns for week days and Number of weeks

daily_activity['Day_of_the_week'] = daily_activity['ActivityDate'].dt.day_name()
daily_activity['Number_of_week'] = daily_activity['ActivityDate'].dt.weekday

#### Extract Month Names

In [166]:
#Extract month names

daily_activity['Month_name'] = daily_activity['ActivityDate'].dt.month_name()

In [167]:
daily_activity.head(10)  #(0: Monday, 1: Tuesday and so on)

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366   2016-03-25       11004           7.11             7.11   
1  1503960366   2016-03-26       17609          11.55            11.55   
2  1503960366   2016-03-27       12736           8.53             8.53   
3  1503960366   2016-03-28       13231           8.93             8.93   
4  1503960366   2016-03-29       12041           7.85             7.85   
5  1503960366   2016-03-30       10970           7.16             7.16   
6  1503960366   2016-03-31       12256           7.86             7.86   
7  1503960366   2016-04-01       12262           7.87             7.87   
8  1503960366   2016-04-02       11248           7.25             7.25   
9  1503960366   2016-04-03       10016           6.37             6.37   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                2.57                      0.46   
1                       0.0                6.92                      0.73   
2                       0.0                4.66                      0.16   
3                       0.0                3.19                      0.79   
4                       0.0                2.16                      1.09   
5                       0.0                2.36                      0.51   
6                       0.0                2.29                      0.49   
7                       0.0                3.32                      0.83   
8                       0.0                3.00                      0.45   
9                       0.0                0.91                      1.28   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 4.07                      0.0                 33   
1                 3.91                      0.0                 89   
2                 3.71                      0.0                 56   
3                 4.95                      0.0                 39   
4                 4.61                      0.0                 28   
5                 4.29                      0.0                 30   
6                 5.04                      0.0                 33   
7                 3.64                      0.0                 47   
8                 3.74                      0.0                 40   
9                 4.18                      0.0                 15   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                   12                   205               804      1819   
1                   17                   274               588      2154   
2                    5                   268               605      1944   
3                   20                   224              1080      1932   
4                   28                   243               763      1886   
5                   13                   223              1174      1820   
6                   12                   239               820      1889   
7                   21                   200               866      1868   
8                   11                   244               636      1843   
9                   30                   314               655      1850   

  Day_of_the_week  Number_of_week Month_name  
0          Friday               4      March  
1        Saturday               5      March  
2          Sunday               6      March  
3          Monday               0      March  
4         Tuesday               1      March  
5       Wednesday               2      March  
6        Thursday               3      March  
7          Friday               4      April  
8        Saturday               5      April  
9          Sunday               6      April

# Heartrate_seconds

In [168]:
heartrate_seconds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3638339 entries, 0 to 3638338
Data columns (total 3 columns):
 #   Column  Dtype         
---  ------  -----         
 0   Id      int64         
 1   Time    datetime64[ns]
 2   Value   int64         
dtypes: datetime64[ns](1), int64(2)
memory usage: 83.3 MB


In [170]:
heartrate_seconds['Time']

0         2016-04-01 07:54:00
1         2016-04-01 07:54:05
2         2016-04-01 07:54:10
3         2016-04-01 07:54:15
4         2016-04-01 07:54:20
                  ...        
3638334   2016-05-12 14:43:53
3638335   2016-05-12 14:43:58
3638336   2016-05-12 14:44:03
3638337   2016-05-12 14:44:18
3638338   2016-05-12 14:44:28
Name: Time, Length: 3638339, dtype: datetime64[ns]

#### Extract Day of the week, week number, hour and month

In [171]:
#Extract week days, week number, month and hour

heartrate_seconds['Day_of_the_week'] = heartrate_seconds['Time'].dt.day_name()
heartrate_seconds['Number_of_week'] = heartrate_seconds['Time'].dt.weekday
heartrate_seconds['Month_name'] = heartrate_seconds['Time'].dt.month_name()
heartrate_seconds['Hour_of_the_day'] = heartrate_seconds['Time'].dt.hour

In [172]:
heartrate_seconds.head(10)

Id                Time  Value Day_of_the_week  Number_of_week  \
0  2022484408 2016-04-01 07:54:00     93          Friday               4   
1  2022484408 2016-04-01 07:54:05     91          Friday               4   
2  2022484408 2016-04-01 07:54:10     96          Friday               4   
3  2022484408 2016-04-01 07:54:15     98          Friday               4   
4  2022484408 2016-04-01 07:54:20    100          Friday               4   
5  2022484408 2016-04-01 07:54:25    101          Friday               4   
6  2022484408 2016-04-01 07:54:30    104          Friday               4   
7  2022484408 2016-04-01 07:54:35    105          Friday               4   
8  2022484408 2016-04-01 07:54:45    102          Friday               4   
9  2022484408 2016-04-01 07:54:55    106          Friday               4   

  Month_name  Hour_of_the_day  
0      April                7  
1      April                7  
2      April                7  
3      April                7  
4      April                7  
5      April                7  
6      April                7  
7      April                7  
8      April                7  
9      April                7

# hourly_cal_int_step


In [173]:
hourly_cal_int_step.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 46183 entries, 0 to 46182
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Id                46183 non-null  int64         
 1   ActivityHour      46183 non-null  datetime64[ns]
 2   Calories          46183 non-null  int64         
 3   StepTotal         46183 non-null  int64         
 4   TotalIntensity    46183 non-null  int64         
 5   AverageIntensity  46183 non-null  float64       
dtypes: datetime64[ns](1), float64(1), int64(4)
memory usage: 2.1 MB


In [174]:
hourly_cal_int_step['ActivityHour']

0       2016-03-12 00:00:00
1       2016-03-12 01:00:00
2       2016-03-12 02:00:00
3       2016-03-12 03:00:00
4       2016-03-12 04:00:00
                ...        
46178   2016-05-12 10:00:00
46179   2016-05-12 11:00:00
46180   2016-05-12 12:00:00
46181   2016-05-12 13:00:00
46182   2016-05-12 14:00:00
Name: ActivityHour, Length: 46183, dtype: datetime64[ns]

#### Extract Day_of_the_week,   week_number, month, hour

In [175]:
hourly_cal_int_step['Day_of_the_week'] = hourly_cal_int_step['ActivityHour'].dt.day_name()
hourly_cal_int_step['Number_of_week'] = hourly_cal_int_step['ActivityHour'].dt.weekday
hourly_cal_int_step['Month_name'] = hourly_cal_int_step['ActivityHour'].dt.month_name()
hourly_cal_int_step['Hour_of_the_day'] = hourly_cal_int_step['ActivityHour'].dt.hour

In [176]:
hourly_cal_int_step.head(10)

Id        ActivityHour  Calories  StepTotal  TotalIntensity  \
0  1503960366 2016-03-12 00:00:00        48          0               0   
1  1503960366 2016-03-12 01:00:00        48          0               0   
2  1503960366 2016-03-12 02:00:00        48          0               0   
3  1503960366 2016-03-12 03:00:00        48          0               0   
4  1503960366 2016-03-12 04:00:00        48          0               0   
5  1503960366 2016-03-12 05:00:00        48          0               0   
6  1503960366 2016-03-12 06:00:00        48          0               0   
7  1503960366 2016-03-12 07:00:00        48          0               0   
8  1503960366 2016-03-12 08:00:00        48          0               0   
9  1503960366 2016-03-12 09:00:00        49          8               1   

   AverageIntensity Day_of_the_week  Number_of_week Month_name  \
0          0.000000        Saturday               5      March   
1          0.000000        Saturday               5      March   
2          0.000000        Saturday               5      March   
3          0.000000        Saturday               5      March   
4          0.000000        Saturday               5      March   
5          0.000000        Saturday               5      March   
6          0.000000        Saturday               5      March   
7          0.000000        Saturday               5      March   
8          0.000000        Saturday               5      March   
9          0.016667        Saturday               5      March   

   Hour_of_the_day  
0                0  
1                1  
2                2  
3                3  
4                4  
5                5  
6                6  
7                7  
8                8  
9                9

# min_cal_int_met_steps


In [212]:
min_cal_int_met_steps.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2770620 entries, 0 to 2770619
Data columns (total 7 columns):
 #   Column           Dtype         
---  ------           -----         
 0   Id               int64         
 1   ActivityMinute   object        
 2   Calories         float64       
 3   ActitvityMinute  datetime64[ns]
 4   Intensity        int64         
 5   METs             int64         
 6   Steps            int64         
dtypes: datetime64[ns](1), float64(1), int64(4), object(1)
memory usage: 148.0+ MB


In [215]:
min_cal_int_met_steps.tail(10)

Id        ActivityMinute  Calories     ActitvityMinute  \
2770610  8877689391  5/12/2016 1:50:00 PM   1.33353 2016-05-12 13:50:00   
2770611  8877689391  5/12/2016 1:51:00 PM   1.33353 2016-05-12 13:51:00   
2770612  8877689391  5/12/2016 1:52:00 PM   1.33353 2016-05-12 13:52:00   
2770613  8877689391  5/12/2016 1:53:00 PM   1.33353 2016-05-12 13:53:00   
2770614  8877689391  5/12/2016 1:54:00 PM   1.33353 2016-05-12 13:54:00   
2770615  8877689391  5/12/2016 1:55:00 PM   1.33353 2016-05-12 13:55:00   
2770616  8877689391  5/12/2016 1:56:00 PM   1.33353 2016-05-12 13:56:00   
2770617  8877689391  5/12/2016 1:57:00 PM   1.33353 2016-05-12 13:57:00   
2770618  8877689391  5/12/2016 1:58:00 PM   1.33353 2016-05-12 13:58:00   
2770619  8877689391  5/12/2016 1:59:00 PM   1.33353 2016-05-12 13:59:00   

         Intensity  METs  Steps  
2770610          0    11      0  
2770611          0    11      0  
2770612          0    11      0  
2770613          0    11      0  
2770614          0    11      0  
2770615          0    11      0  
2770616          0    11      0  
2770617          0    11      0  
2770618          0    11      0  
2770619          0    11      0

#### Rename the 'ActitvityMinute' with 'ActivityMinute_24hr'

In [216]:
min_cal_int_met_steps.rename(columns={'ActitvityMinute' : 'ActivityMinute_24hr'}, inplace = True)

#### Drop "ActivityMinute" column 

In [218]:
min_cal_int_met_steps.drop(columns =['ActivityMinute'], inplace = True)

In [219]:
min_cal_int_met_steps.head(10)

Id  Calories ActivityMinute_24hr  Intensity  METs  Steps
0  1503960366    0.7973 2016-03-12 00:00:00          0    10      0
1  1503960366    0.7973 2016-03-12 00:01:00          0    10      0
2  1503960366    0.7973 2016-03-12 00:02:00          0    10      0
3  1503960366    0.7973 2016-03-12 00:03:00          0    10      0
4  1503960366    0.7973 2016-03-12 00:04:00          0    10      0
5  1503960366    0.7973 2016-03-12 00:05:00          0    10      0
6  1503960366    0.7973 2016-03-12 00:06:00          0    10      0
7  1503960366    0.7973 2016-03-12 00:07:00          0    10      0
8  1503960366    0.7973 2016-03-12 00:08:00          0    10      0
9  1503960366    0.7973 2016-03-12 00:09:00          0    10      0

#### Extract day of the week, week number, month, hour and minute

In [220]:
min_cal_int_met_steps.columns

Index(['Id', 'Calories', 'ActivityMinute_24hr', 'Intensity', 'METs', 'Steps'], dtype='object')

In [222]:
min_cal_int_met_steps['Day_of_the_week'] = min_cal_int_met_steps['ActivityMinute_24hr'].dt.day_name()
min_cal_int_met_steps['Number_of_week'] = min_cal_int_met_steps['ActivityMinute_24hr'].dt.weekday
min_cal_int_met_steps['Month_name'] = min_cal_int_met_steps['ActivityMinute_24hr'].dt.month_name()
min_cal_int_met_steps['Minute_of_the_hour'] = min_cal_int_met_steps['ActivityMinute_24hr'].dt.minute
min_cal_int_met_steps['Hour_of_the_day'] = min_cal_int_met_steps['ActivityMinute_24hr'].dt.hour

In [236]:
min_cal_int_met_steps.tail(10)

Id  Calories ActivityMinute_24hr  Intensity  METs  Steps  \
2770610  8877689391   1.33353 2016-05-12 13:50:00          0    11      0   
2770611  8877689391   1.33353 2016-05-12 13:51:00          0    11      0   
2770612  8877689391   1.33353 2016-05-12 13:52:00          0    11      0   
2770613  8877689391   1.33353 2016-05-12 13:53:00          0    11      0   
2770614  8877689391   1.33353 2016-05-12 13:54:00          0    11      0   
2770615  8877689391   1.33353 2016-05-12 13:55:00          0    11      0   
2770616  8877689391   1.33353 2016-05-12 13:56:00          0    11      0   
2770617  8877689391   1.33353 2016-05-12 13:57:00          0    11      0   
2770618  8877689391   1.33353 2016-05-12 13:58:00          0    11      0   
2770619  8877689391   1.33353 2016-05-12 13:59:00          0    11      0   

        Day_of_the_week  Number_of_week Month_name  Minute_of_the_hour  \
2770610        Thursday               3        May                  50   
2770611        Thursday               3        May                  51   
2770612        Thursday               3        May                  52   
2770613        Thursday               3        May                  53   
2770614        Thursday               3        May                  54   
2770615        Thursday               3        May                  55   
2770616        Thursday               3        May                  56   
2770617        Thursday               3        May                  57   
2770618        Thursday               3        May                  58   
2770619        Thursday               3        May                  59   

         Hour_of_the_day  
2770610               13  
2770611               13  
2770612               13  
2770613               13  
2770614               13  
2770615               13  
2770616               13  
2770617               13  
2770618               13  
2770619               13

In [237]:
min_cal_int_met_steps['Minute_of_the_hour'].unique()

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50,
       51, 52, 53, 54, 55, 56, 57, 58, 59], dtype=int64)

In [238]:
min_cal_int_met_steps['Hour_of_the_day'].unique()

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23], dtype=int64)

In [239]:
min_cal_int_met_steps['Month_name'].unique()

array(['March', 'April', 'May'], dtype=object)

# minute_sleep


In [240]:
minute_sleep.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 387080 entries, 0 to 387079
Data columns (total 4 columns):
 #   Column  Non-Null Count   Dtype         
---  ------  --------------   -----         
 0   Id      387080 non-null  int64         
 1   date    387080 non-null  datetime64[ns]
 2   value   387080 non-null  int64         
 3   logId   387080 non-null  int64         
dtypes: datetime64[ns](1), int64(3)
memory usage: 11.8 MB


In [241]:
minute_sleep.head(10)

Id                date  value        logId
0  1503960366 2016-03-13 02:39:30      1  11114919637
1  1503960366 2016-03-13 02:40:30      1  11114919637
2  1503960366 2016-03-13 02:41:30      1  11114919637
3  1503960366 2016-03-13 02:42:30      1  11114919637
4  1503960366 2016-03-13 02:43:30      1  11114919637
5  1503960366 2016-03-13 02:44:30      1  11114919637
6  1503960366 2016-03-13 02:45:30      2  11114919637
7  1503960366 2016-03-13 02:46:30      2  11114919637
8  1503960366 2016-03-13 02:47:30      1  11114919637
9  1503960366 2016-03-13 02:48:30      1  11114919637

#### Extract day of the week, number of week, month_name, hour_of_the_day

In [242]:
minute_sleep['Day_of_the_week'] = minute_sleep['date'].dt.day_name()
minute_sleep['Number_of_week'] = minute_sleep['date'].dt.weekday
minute_sleep['Month_name'] = minute_sleep['date'].dt.month_name()
minute_sleep['Hour_of_the_day'] = minute_sleep['date'].dt.hour

In [243]:
minute_sleep.head(10)

Id                date  value        logId Day_of_the_week  \
0  1503960366 2016-03-13 02:39:30      1  11114919637          Sunday   
1  1503960366 2016-03-13 02:40:30      1  11114919637          Sunday   
2  1503960366 2016-03-13 02:41:30      1  11114919637          Sunday   
3  1503960366 2016-03-13 02:42:30      1  11114919637          Sunday   
4  1503960366 2016-03-13 02:43:30      1  11114919637          Sunday   
5  1503960366 2016-03-13 02:44:30      1  11114919637          Sunday   
6  1503960366 2016-03-13 02:45:30      2  11114919637          Sunday   
7  1503960366 2016-03-13 02:46:30      2  11114919637          Sunday   
8  1503960366 2016-03-13 02:47:30      1  11114919637          Sunday   
9  1503960366 2016-03-13 02:48:30      1  11114919637          Sunday   

   Number_of_week Month_name  Hour_of_the_day  
0               6      March                2  
1               6      March                2  
2               6      March                2  
3               6      March                2  
4               6      March                2  
5               6      March                2  
6               6      March                2  
7               6      March                2  
8               6      March                2  
9               6      March                2

In [244]:
minute_sleep['value'].unique()

array([1, 2, 3], dtype=int64)

In [245]:
minute_sleep['Hour_of_the_day'].unique()

array([ 2,  3,  4,  5,  6,  7,  8,  9,  1, 19, 20,  0, 23, 10, 11, 15, 16,
       17, 18, 21, 22, 12, 13, 14], dtype=int64)

In [249]:
minute_sleep.dtypes

Id                          int64
date               datetime64[ns]
value                       int64
logId                       int64
Day_of_the_week            object
Number_of_week              int64
Month_name                 object
Hour_of_the_day             int64
dtype: object

# weight_log_info

In [250]:
weight_log_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              100 non-null    int64  
 1   Date            100 non-null    object 
 2   WeightKg        100 non-null    float64
 3   WeightPounds    100 non-null    float64
 4   Fat             4 non-null      float64
 5   BMI             100 non-null    float64
 6   IsManualReport  100 non-null    bool   
 7   LogId           100 non-null    int64  
dtypes: bool(1), float64(4), int64(2), object(1)
memory usage: 5.7+ KB


In [252]:
weight_log_info['Id'].nunique()

13

### Save the files for analysis

In [253]:
daily_activity.to_csv("daily_activity.csv", index = False)

heartrate_seconds.to_csv("heartrate_seconds.csv", index = False)

hourly_cal_int_step.to_csv("hourly_cal_int_step.csv", index = False)

min_cal_int_met_steps.to_csv("minutes_cal_int_MET_steps.csv", index = False)

minute_sleep.to_csv("minute_sleep.csv", index = False)

weight_log_info.to_csv("weight_log_info.csv", index = False)